### ISSUE TITLE
tblis fails to compile

### ISSUE THREAD

**Issue by h-larsson** [pyscf/pyscf#502]
After the recent changes in https://github.com/pyscf/pyscf/commit/fb9e38f479c2807783b4ff5e8c1df6874a3e21fe, I get the following error during the build with tblis:
```
[ 24%] Linking CXX shared library ../libtblis_einsum.so
/usr/bin/ld: cannot find -ltblis
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/tblis_einsum.dir/build.make:84: ../libtblis_einsum.so] Error 1
make[1]: *** [CMakeFiles/Makefile2:462: CMakeFiles/tblis_einsum.dir/all] Error 2
```
libtblis.so is in 
```./deps/src/libtblis-build/lib/.libs/libtblis.so```

**Comment by sunqm** [comment]
Problem should be fixed in c41e7e9

**Comment by h-larsson** [comment]
Seems to work. However, I have to run ```make``` twice. In the first run, I get the following error:
```
[ 19%] Building CXX object CMakeFiles/tblis_einsum.dir/np_helper/as_einsum.cxx.o
/home/larsson/software/pyscf/pyscf/lib/np_helper/as_einsum.cxx:2:10: fatal error: tblis/tblis.h: No such file or directory
    2 | #include <tblis/tblis.h>
      |          ^~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/tblis_einsum.dir/build.make:63: CMakeFiles/tblis_einsum.dir/np_helper/as_einsum.cxx.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:491: CMakeFiles/tblis_einsum.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```
In the second run, it compiles.

**Comment by yangcal** [comment]
> Seems to work. However, I have to run `make` twice. In the first run, I get the following error:
> 
> ```
> [ 19%] Building CXX object CMakeFiles/tblis_einsum.dir/np_helper/as_einsum.cxx.o
> /home/larsson/software/pyscf/pyscf/lib/np_helper/as_einsum.cxx:2:10: fatal error: tblis/tblis.h: No such file or directory
>     2 | #include <tblis/tblis.h>
>       |          ^~~~~~~~~~~~~~~
> compilation terminated.
> make[2]: *** [CMakeFiles/tblis_einsum.dir/build.make:63: CMakeFiles/tblis_einsum.dir/np_helper/as_einsum.cxx.o] Error 1
> make[1]: *** [CMakeFiles/Makefile2:491: CMakeFiles/tblis_einsum.dir/all] Error 2
> make[1]: *** Waiting for unfinished jobs....
> ```
> 
> In the second run, it compiles.

I found similar issue. The Makefile seems to have incorrect order for libtblis and tblis_einsum. I had to first make libtblis and then it will compile.

**Comment by sunqm** [comment]
The compiling dependency issue was fixed in PR #534
